tests/test_symbol.py

```diff
@@ -205,3 +218,27 @@ def test_kind(self):
                     (ElfSymbol("foo", 0xFFFF0000, 1, elf_type, STB.GLOBAL),)
                 )
                 self.assertEqual(prog.symbol("foo").kind, drgn_kind)
+                symbols = prog.symbols("foo")
+                self.assertEqual(len(symbols), 1)
+                self.assertEqual(symbols[0].kind, drgn_kind)
+
+    def test_all_symbols(self):
+        elf_syms = (
+            (
+                ElfSymbol("two", 0xFFF11000, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("three", 0xFFF1200, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("one", 0xFFF10000, 1, STT.OBJECT, STB.GLOBAL),
+            ),
+            (
+                ElfSymbol("three", 0xFFF21000, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("two", 0xFFF20000, 1, STT.OBJECT, STB.GLOBAL),
+            ),
+            (ElfSymbol("three", 0xFFF30000, 1, STT.OBJECT, STB.GLOBAL),),
+        )
+        prog = elf_symbol_program(*elf_syms)
+        symbols = prog.symbols()
+        self.assertEqual(len(symbols), 6)
+        names = Counter(sym.name for sym in symbols)
+        self.assertEqual(names["one"], 1)
+        self.assertEqual(names["two"], 2)
+        self.assertEqual(names["three"], 3)
```

Comment:
The test would be more thorough if you sorted the returned symbols in some well-defined order and used the aforementioned `assert_symbols_equal`. (Also make sure to drop the `Counter` import if you do that.)